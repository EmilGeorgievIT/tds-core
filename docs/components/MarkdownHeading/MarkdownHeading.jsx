import React from 'react'
import PropTypes from 'prop-types'

import Box from '../../../src/components/Box/Box'
import Heading from '../../../src/components/Typography/Heading/Heading'

const MarkdownHeading = ({spacing, level, children}) => (
  <Box spacing="margin" bottom={spacing}>
    <Heading level={level}>{children}</Heading>
  </Box>
)

MarkdownHeading.propTypes = {
  spacing: PropTypes.number.isRequired,
  level: PropTypes.string.isRequired,
  children: PropTypes.string.isRequired,
}

export default MarkdownHeading
