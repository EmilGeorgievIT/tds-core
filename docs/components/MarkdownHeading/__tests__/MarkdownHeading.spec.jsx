import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'

import MarkdownHeading from '../MarkdownHeading'

describe('MarkdownHeading', () => {
  it('renders', () => {
    const markdownHeading = shallow(
      <MarkdownHeading level="h1" spacing={3}>
        Random Heading
      </MarkdownHeading>
    )

    expect(toJson(markdownHeading)).toMatchSnapshot()
  })
})
