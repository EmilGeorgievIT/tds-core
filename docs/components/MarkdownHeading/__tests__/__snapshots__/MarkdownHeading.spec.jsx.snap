// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkdownHeading renders 1`] = `
<Box
  bottom={3}
  inline={false}
  spacing="margin"
>
  <Heading
    invert={false}
    level="h1"
  >
    Random Heading
  </Heading>
</Box>
`;
